cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -62,19 +63,25 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.MasterC
 		errs = append(errs, err)
 	}
 
-	// Create/update RBAC rules that makes the bootstrap tokens able to get their CSRs approved automatically
-	if err := nodebootstraptoken.AutoApproveNodeBootstrapTokens(client, k8sVersion); err != nil {
-		errs = append(errs, err)
+	// Not needed for 1.7 upgrades
+	if k8sVersion.AtLeast(constants.MinimumCSRAutoApprovalClusterRolesVersion) {
+		// Create/update RBAC rules that makes the bootstrap tokens able to get their CSRs approved automatically
+		if err := nodebootstraptoken.AutoApproveNodeBootstrapTokens(client, k8sVersion); err != nil {
+			errs = append(errs, err)
+		}
 	}
 
 	// TODO: Is this needed to do here? I think that updating cluster info should probably be separate from a normal upgrade
 	// Create the cluster-info ConfigMap with the associated RBAC rules
 	// if err := clusterinfo.CreateBootstrapConfigMapIfNotExists(client, kubeadmconstants.GetAdminKubeConfigPath()); err != nil {
 	// 	return err
 	//}
-	// Create/update RBAC rules that makes the cluster-info ConfigMap reachable
-	if err := clusterinfo.CreateClusterInfoRBACRules(client); err != nil {
-		errs = append(errs, err)
+	// Not needed for 1.7 upgrades
+	if k8sVersion.AtLeast(constants.UseEnableBootstrapTokenAuthFlagVersion) {
```

Comment:
this is incorrect... we still need this permission set up for 1.7 installs

Reply:
It exists if 1.7.x was initialized with kubeadm. Only with different roleRef than expected in 1.8.
Not running it, means cluster will be in the same state as it was expected within 1.7.x (no role/bindings migrations to 1.8 naming schema).

That's what @luxas suggested. First version of that PR had migration to 1.8 naming schema even in cases of 1.7.x->1.7.x upgrade.

Reply:
ah, this is just the upgrade path. this still seems like a fragile way to determine what changes are needed, but I'll defer to @luxas.

were all currently supported upgrade paths tested with this change?
1.7.x -> 1.7.y upgrade
1.7.x -> 1.8.x upgrade
1.8.0 -> 1.8.0 "upgrade"

Reply:
Yes, I've tested following on my development cluster:
- install fresh cluster version 1.7.3 using kubeadm 1.7.5
  - old roles/bindings naming schema
- upgrade 1.7.3 to 1.7.7 using modified kubeadm 1.8.0 
  - no roles/binding changes, old names/values
- upgrade 1.7.7 to 1.8.0-rc.1 using modified kubeadm 1.8.0 
  - roles/bindings migrated to 1.8 schema
- upgrade 1.8.0-rc.1 to 1.8.0 using modified kubeadm 1.8.0
  - roles/bindings in 1.8 naming schema